File: LinkedList/simply/python/LinkedList/LinkedList.py
class Node():
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList():
    def __init__(self):
        self.head = None
        self.lastNode = None

    def add(self, data):

        if self.head == None:
            self.head = Node(data)
            self.lastNode = self.head
            
            return

        self.lastNode.next = Node(data)
        self.lastNode = self.lastNode.next

    def printAll(self):
        tmp = self.head

        while tmp:
            print(tmp.data)
            tmp = tmp.next

File: LinkedList/simply/javascript/LinkedList/LinkedList.js
function Node(data) {
	this.data = data;
	this.next = null;
}

exports.LinkedList = function() {
	this.head = null;
	this.lastNode = null;

	this.push = function(data) {
		if(this.head === null) {
			this.head = new Node(data);
			this.lastNode = this.head;

			return;
		}

		this.lastNode.next = new Node(data);
		this.lastNode = this.lastNode.next;
	}

	this.shift = function(data) {
		if(this.head === null) {
			this.head = new Node(data);
			this.lastNode = this.head;

			return;
		}

		let tmp = this.head;
		this.head = new Node(data);
		this.head.next = tmp;
	}

	this.printAll = function() {
		tmp = this.head;
		while(tmp) {
			console.log(tmp.data);
			tmp = tmp.next;
		}
	}
}

